### DISCUSSION TITLE
Why is the mount event of the child component executed twice?

### DISCUSSION THREAD

**Discussion by lzm0x219** [solidjs/solid#385] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/50414099/112782258-2d49f500-907f-11eb-93f4-d8258facb3e0.png)

**Comment by ryansolid** [comment, marked as answer]
I'm going to need to see more than this. It doesn't fire twice generally so I suspect something else is going on.

**Reply by lzm0x219** [reply]
[example](https://github.com/vvni/solid-f2)。The problem can be reproduced here.

**Reply by lzm0x219** [reply]
Are there any problems found?

**Reply by ryansolid** [reply]
I found it. Children are being created multiple times.
```js
chart = new AntvChart({
        el: ref,
        ...chartParams,
      });
```
Accesses all the lazy evalled props (including children) in the spread as `chartParams` has `children` on it.

I fixed it by splitting children out:
```jsx
const [canvasProps, _, chartProps] = splitProps(props, ["class", "className"], ["children"]);
```

**Reply by lzm0x219** [reply]
thanks.~
